content/latest/api/ysql/commands/ddl_create_domain.md

```diff
@@ -0,0 +1,71 @@
+
+#### domain_constraint
+
+<svg class="rrdiagram" version="1.1" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns="http://www.w3.org/2000/svg" width="634" height="125" viewbox="0 0 634 125"><path class="connector" d="M0 37h25m98 0h10m122 0h20m-265 0q5 0 5 5v8q0 5 5 5h240q5 0 5-5v-8q0-5 5-5m5 0h50m-5 0q-5 0-5-5v-17q0-5 5-5h274q5 0 5 5v17q0 5-5 5m-269 0h20m45 0h10m52 0h137m-254 25q0 5 5 5h5m52 0h177q5 0 5-5m-249-25q5 0 5 5v50q0 5 5 5h5m61 0h10m25 0h10m83 0h10m25 0h5q5 0 5-5v-50q0-5 5-5m5 0h40m-339 0q5 0 5 5v68q0 5 5 5h314q5 0 5-5v-68q0-5 5-5m5 0h5"/><rect class="literal" x="25" y="20" width="98" height="25" rx="7"/><text class="text" x="35" y="37">CONSTRAINT</text><a xlink:href="../grammar_diagrams#constraint-name"><rect class="rule" x="133" y="20" width="122" height="25"/><text class="text" x="143" y="37">constraint_name</text></a><rect class="literal" x="345" y="20" width="45" height="25" rx="7"/><text class="text" x="355" y="37">NOT</text><rect class="literal" x="400" y="20" width="52" height="25" rx="7"/><text class="text" x="410" y="37">NULL</text><rect class="literal" x="345" y="50" width="52" height="25" rx="7"/><text class="text" x="355" y="67">NULL</text><rect class="literal" x="345" y="80" width="61" height="25" rx="7"/><text class="text" x="355" y="97">CHECK</text><rect class="literal" x="416" y="80" width="25" height="25" rx="7"/><text class="text" x="426" y="97">(</text><a xlink:href="../grammar_diagrams#expression"><rect class="rule" x="451" y="80" width="83" height="25"/><text class="text" x="461" y="97">expression</text></a><rect class="literal" x="544" y="80" width="25" height="25" rx="7"/><text class="text" x="554" y="97">)</text></svg>
+
+### Grammar
+```
+create_domain ::= CREATE DOMAIN name [ AS ] data_type
+    [ COLLATE collation ]
+    [ DEFAULT expression ]
+    [ domain_constraint [ ... ] ];
+
+domain_constraint ::= [ CONSTRAINT constraint_name ]
+{ NOT NULL | NULL | CHECK (expression) };
+```
+
+Where
+
+- `name` is the name of the domain.
+- `data_type` is the underlying data type.
+- `COLLATE collation` is an optional collation for the domain.
+- `DEFAULT expression` sets default value for columns and must match data type of the domain.
+- `CONSTRAINT constraint_name` is an optional name for constraint.
+- `NOT NULL` does not allow null values.
+- `NULL` allows null values (default).
+- `CHECK (expression)` must produce a boolean result.
```

Comment:
Add that "This enforces a constraint that the values of the domain must satisfy. You should use the key word VALUE to refer to the value being tested. Expressions evaluating to TRUE or UNKNOWN succeed."